src/session.rs

```diff
@@ -72,10 +128,168 @@ impl RequestResult {
     }
 }
 
+#[pyclass]
+#[derive(Clone)]
+pub(crate) struct PyRowSerializationContext {
+    prepared_statement: Arc<scylla::statement::prepared::PreparedStatement>,
+}
+
+#[pymethods]
+impl PyRowSerializationContext {
+    /// Create a new PyRowSerializationContext from a PyPreparedStatement
+    #[staticmethod]
+    fn from_prepared(prepared: PyPreparedStatement) -> Self {
+        Self {
+            prepared_statement: Arc::clone(&prepared._inner),
+        }
+    }
+
+    /// Get column specifications as a list of dictionaries
+    /// This provides Python access to column metadata
+    fn columns(&self) -> PyResult<Vec<PyObject>> {
+        Python::with_gil(|py| {
+            let metadata = self.prepared_statement.get_prepared_metadata();
+            let context = RowSerializationContext::from_prepared(metadata);
+
+            let mut columns = Vec::new();
+            for col_spec in context.columns() {
+                let col_dict = pyo3::types::PyDict::new(py);
+
+                // Add column name
+                col_dict.set_item("name", col_spec.name())?;
+
+                // Extract structured type information
+                let type_info = extract_type_info(py, col_spec.typ())?;
+                col_dict.set_item("type_struct", type_info)?;
+
+                columns.push(col_dict.into());
+            }
+
+            Ok(columns)
+        })
+    }
+}
+
+// Create python dict
+// kind = ColumnType
+// ...
+// if kind: ColumnType != Native
+// element_type = dict - recursively calll this method
+fn extract_type_info(py: Python, column_type: &ColumnType) -> PyResult<PyObject> {
+    let type_dict = pyo3::types::PyDict::new(py);
+
+    match column_type {
+        // POC: Handle Native types with full structure
+        ColumnType::Native(native_type) => {
+            type_dict.set_item("kind", "native")?;
```

Comment:
💭 Although this might seem like redundant overhead (compared to solutions where we serialize in Rust), let's keep in mind that we will probably want to expose prepared metadata (and result metadata too, which is analogous for deserialization case) to Python user anyway.

So such API will be needed no matter which serialization flavour we go for. One important thing: if we decide to serialize in Rust, we will be able to make this type info extraction lazy, i.e., executed only on user explicit demand.